packages/components/src/components/send-email-template/send-email-template.jsx

```diff
@@ -3,88 +3,92 @@ import React from 'react';
 import Button from '../button/button.jsx';
 import Icon from '../icon/icon.jsx';
 
-class SendEmailTemplate extends React.Component {
-    resend_interval = null;
+const SendEmailTemplate = ({
+    children,
+    className,
+    lbl_no_receive,
+    onClickSendEmail,
+    resend_timeout,
+    subtitle,
+    title,
+    txt_resend,
+    txt_resend_in,
+}) => {
+    const [is_email_not_received_clicked, setIsEmailNotReceivedClicked] = React.useState(false);
+    const [is_resend_btn_disabled, setIsResendBtnDisabled] = React.useState(false);
+    const [resend_email_btn_text, setResendEmailBtnText] = React.useState(txt_resend);
 
-    state = {
-        is_email_not_received_clicked: false,
-        resend_timeout: this.props.resend_timeout || 60,
-    };
+    const timeout_limit = resend_timeout || 60;
+    let resend_interval = null;
 
-    componentWillUnmount() {
-        clearInterval(this.resend_interval);
-    }
+    React.useEffect(() => {
+        return () => {
+            clearInterval(resend_interval);
+        };
+    }, []);
```

Comment:
does this lead to an exhaustive depth Eslint warning?

Reply:
As per [this](https://stackoverflow.com/questions/56972415/useeffect-dependency-array-and-eslint-exhaustive-deps-rule) I might want to put `resend_interval` as a dependency to avoid this warning. But my concern is that we if `resend_interval` is being changes will that affect the component on `unmount`?

Reply:
By right, every time `resend_interval` changes, this useEffect is re-rending, so my suggestion is that you add a rule to disable Eslint warning.